src/System.Data.SqlClient/src/System/Data/Common/DbConnectionOptions.cs

```diff
@@ -75,7 +75,7 @@ internal bool ConvertValueToIntegratedSecurityInternal(string stringValue)
         public int ConvertValueToInt32(string keyName, int defaultValue)
         {
             string value;
-            return _parsetable.TryGetValue(keyName, out value) ?
+            return _parsetable.TryGetValue(keyName, out value) && !string.IsNullOrWhiteSpace(value) ?
```

Comment:
@stephentoub Indeed, changed to a simple null-check to match reference sources behavior - tests still passed.